CRM/Contract/Form/Modify.php

```diff
@@ -11,6 +11,13 @@ class CRM_Contract_Form_Modify extends CRM_Core_Form{
 
   function preProcess(){
 
+    // If we requested a contract file download
+    $download = CRM_Utils_Request::retrieve('ct_dl', 'String', CRM_Core_DAO::$_nullObject, FALSE, '', 'GET');
+    if (!empty($download)) {
+      CRM_Contract_Utils::downloadContractFile($download);
```

Comment:
@systopia I didn't know CRM_Utils_System::download existed.  However, now I tested it you don't seem to be able to set the downloaded filename and you get a file with the full pathname defined (eg. var-www-httpdocs...-1234.tif).  With my download method you just get a file called 1234.tif.